src/request.ts

```diff
@@ -1330,45 +1325,53 @@ export type ProjectId = string | null | undefined;
 export interface RequestCallback {
   (
     a?: Error | null,
-    b?: AllocateIdsRequestResponse &
+    b?: google.datastore.v1.IAllocateIdsResponse &
       google.datastore.v1.ILookupResponse &
-      Entities
+      google.datastore.v1.ICommitResponse &
+      google.datastore.v1.IRollbackResponse & {
+        transaction?: string;
+      }
   ): void;
 }
 export interface RequestConfig {
   client: string;
-  gaxOpts?: number | CallOptions | KeyProto;
+  gaxOpts?: CallOptions;
   method: string;
   prepared?: boolean;
-  reqOpts?: Entity | RequestOptions;
-  gaxOptions?: never;
+  reqOpts?: RequestOptions;
 }
 export interface RequestOptions {
-  mutations?: [] | Array<{delete: KeyProto}> | Array<{}>;
+  mutations?: google.datastore.v1.IMutation[];
   keys?: Entity;
-  readOptions?: {readConsistency?: number; transaction?: string | number};
+  readOptions?: {readConsistency?: number; transaction?: string};
+  partitionId?: google.datastore.v1.IPartitionId | null;
   transactionOptions?: {
     readOnly?: {};
     readWrite?: {previousTransaction?: string};
-  };
-  transaction?: string | number;
+  } | null;
+  transaction?: string | null;
   mode?: string;
-  projectId?: string;
-}
-export interface RunQueryStreamOptions {
-  gaxOptions?: CallOptions;
-  consistency?: 'strong' | 'eventual';
+  projectId?: ProjectId;
+  query?: QueryProto;
 }
-export interface SaveCallback {
-  (a?: Error | null, b?: Entity): void;
+export interface RunQueryStreamOptions extends RunQueryOptions {}
```

Comment:
Any reason why we wouldn't just re-use `RunQueryOptions`?

Reply:
Just to keep the same "aesthetic" look as with `Save`/`Update`/`UpsertCallback` interfaces extend `CommitCallback` interface.
Not particular about it. Can change ether way?

Reply:
No need, I was mostly curious!